### PR TITLE
Feature/pelagos 4148 create method to replace current single file method and get data from files entity

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

File: src/Entity/ResearchGroup.php [hunk 1]
```diff
@@ -353,6 +353,7 @@ public function serializeDatasets()
             } else {
                 $datasetArray['dif'] = null;
             }
+            dump($dataset->getDatasetSubmission());
```
Comment (mickel1138) [inline src/Entity/ResearchGroup.php:356]: dump

File: src/Form/DatasetSubmissionType.php [hunk 2]
```diff
@@ -70,7 +70,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'multiple' => false,
             ))
             ->add('datasetFileUri', Type\HiddenType::class, array(
-                'required' => true,
+                'required' => false,
```
Comment (mickel1138) [inline src/Form/DatasetSubmissionType.php:73]: revert

File: src/Entity/DatasetSubmission.php [hunk 3]
```diff
@@ -1584,7 +1581,7 @@ public function getDatasetFileTransferType() : ?string
      */
     public function setDatasetFileUri(?string $datasetFileUri)
     {
-        $this->datasetFileUri = $datasetFileUri;
+        $this->fileset->getFiles()->first()->setFilePath($datasetFileUri);
```
Comment (praneethpr) [inline src/Entity/DatasetSubmission.php:1584]: Should i revert the changes made to datasetFileUri setters and getters?

Comment (mickel1138) [reply src/Entity/DatasetSubmission.php:1584]: yes, they should stay as they are.

File: src/Entity/DatasetSubmission.php [hunk 4]
```diff
@@ -547,9 +547,6 @@ class DatasetSubmission extends Entity
      *
      * @ORM\Column(type="text", nullable=true)
      *
-     * @Assert\NotBlank(
```
Comment (mickel1138) [inline src/Entity/DatasetSubmission.php:550]: undo

File: src/Entity/DatasetSubmission.php [hunk 5]
```diff
@@ -1594,7 +1591,7 @@ public function setDatasetFileUri(?string $datasetFileUri)
      */
     public function getDatasetFileUri() : ?string
     {
-        return $this->datasetFileUri;
+        return $this->fileset->getFiles()->first()->getFilePath();
     }
 
     /**
```
Comment (mickel1138) [inline src/Entity/DatasetSubmission.php:1597]: Undo also.

File: tests/unit/Util/ISOMetadataExtractorUtilTest.php [hunk 6]
```diff
@@ -224,6 +224,9 @@ public function setUp()
                                 'getRemotelyHostedDescription' => 'remote description',
                                 'getRemotelyHostedFunction' => 'download',
                                 'getDatasetLinks' => new ArrayCollection(),
+                                'getFileset' => \Mockery::mock(
```
Comment (mickel1138) [inline tests/unit/Util/ISOMetadataExtractorUtilTest.php:227]: Nice use of Mockery!

File: src/Entity/DatasetSubmission.php [hunk 7]
```diff
@@ -1020,7 +1020,7 @@ public function __construct(Entity $entity, PersonDatasetSubmissionDatasetContac
             $this->setDatasetFileColdStorageArchiveSha256Hash($entity->getDatasetFileColdStorageArchiveSha256Hash());
             $this->setDatasetFileColdStorageArchiveSize($entity->getDatasetFileColdStorageArchiveSize());
             $this->setDatasetFileColdStorageOriginalFilename($entity->getDatasetFileColdStorageOriginalFilename());
-
+            $this->setFileset($entity->getFileset());
```
Comment (mickel1138) [inline src/Entity/DatasetSubmission.php:1023]: Since this is collection, we have to copy it. Like the foreach things below. The file entities inside may be copied.
I'll look into this, to see if we can make it work.
